src/eduid/webapp/idp/views/next.py

```diff
@@ -276,6 +282,58 @@ def _set_user_options(res: AuthnOptions, eppn: str) -> None:
     return None
 
 
+def _geo_statistics(ticket: LoginContext, sso_session: Optional[SSOSession]) -> None:
+    """ Log user statistics from login event """
+
+    if not sso_session:
+        return None
+
+    if not current_app.conf.geo_statistics_secret_key or not current_app.conf.geo_statistics_url:
+        return None
+
+    ua = get_user_agent()
+    if not ua:
+        return None
+
+    if ua.parsed.browser.family in ['Python Requests', 'PingdomBot'] or ua.parsed.is_bot:
+        return None
+
+    secret = bytes(current_app.conf.geo_statistics_secret_key, 'utf-8')
+
+    user_hash = hmac.HMAC(secret, hashes.SHA256())
+    user_hash.update(bytes(sso_session.eppn, 'utf-8'))
+
+    device_id = None
+    if ticket.known_device:
+        device_id_hash = hmac.HMAC(secret, hashes.SHA256())
+        device_id_hash.update(bytes(ticket.known_device.state_id, 'uft-8'))
+        device_id = device_id_hash.finalize().hex()
```

Comment:
This equates to a 1:1 mapping of database ids in the idp's known_device database, to another identifier I presume you use in the geo statistics system. What kind of privacy/anonymisation/protection/other is it you want to achieve with this?

Reply:
@johanlundberg didn't want eppn to be part of 'geo' db if the application got breached/hacked. To use a hashed version of the eppn works fine in this application since I'm only interested in the relation not the actual person.

Reply:
Maybe we choose the wrong ID? Let's have a discussion about it.